SmartHub/SmartHub/HomeScreen/HomeScreenView.swift

```diff
@@ -0,0 +1,183 @@
+                        } label: {
+                            Label("Add Room", systemImage: "door.left.hand.closed")
+                        }
+                    }
+                    
+                    Divider()
+                    
+                    Button (role: .destructive){
+                        isShowingConfirmation = true
+                    } label: {
+                        Label("Log Out", systemImage:"person.crop.circle")
+                    }
+                    
+                } label: {
+                    Label("more", systemImage: "ellipsis.circle")
+                }
+            }
+            .sheet(isPresented:$isShowingScanner){
+                CodeScannerView(codeTypes: [.qr], completion: handleScan)
+            }
+            .confirmationDialog( "Are you sure you want to log out?", isPresented: $isShowingConfirmation, titleVisibility: .visible) {
+                Button (role: .destructive) {
+                    logout()
+                } label: {
+                    Text ("Log Out")
+                }
+            }
+            .onAppear{
+                homeScreenViewModel.loadAllRooms()
+            }
+        }
+    }
+    
+    func handleScan (result: Result<ScanResult, ScanError>) {
+        isShowingScanner = false
+        
+        switch result {
+        case .success(let result):
+            let details = result.string
+            print(details)
+            
+            //let device = Device(name: details)
+            //homeScreenViewModel.addDeviceToRoom(room: Room.allRooms.first!, device: device)
+            //continue implementation next time
+            //we can crete codes with the data separated by symbol
+        case .failure(_):
+            print("Scanning faild")
+        }
+    }
+    
+    private func logout() {
+        do {
+            try Auth.auth().signOut()
+            viewModel.isLogged = false
+        } catch {
+            print("Error signing out: \(error.localizedDescription)")
+        }
+    }
+}
+
+struct RoomTileView: View {
+    var room: Room
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 8) {
+            Text(room.name)
+                .font(.headline)
+                .foregroundColor(.primary)
+            
+            
+            ForEach(room.devices) { device in
+                DeviceTileView(device: device)
+            }
+        }
+        .padding()
+        .background(Color.secondary.opacity(0.1))
+        .cornerRadius(10)
+        
+    }
+}
+
+struct DeviceTileView: View {
+    var device: Device
+    
+    var body: some View {
+        HStack {
+            Image(systemName: "lightbulb")
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .frame(width: 20, height: 20)
+                .foregroundColor(.blue)
+            
+            Text(device.name)
+                .font(.caption)
+                .foregroundColor(.secondary)
+        }
```

Comment:
Here some of the function arguments should be separated into variables so that the different devices will have different properties